### PR TITLE
Improve controller name definition

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️

### CODE REVIEW

File: src/Controller/AbstractController.php [hunk 1]
```diff
@@ -61,7 +63,7 @@ private function renderContent(string $view, array $parameters = []): string
 
     public function getViewPath(): string
     {
-        return $this->aliases->get('@views') . '/' . $this->name();
+        return $this->aliases->get('@views') . '/' . self::getName();
```
Comment (samdark) [inline src/Controller/AbstractController.php:66]: Better to use static in order to allow changing default algorithm.

File: src/Controller/AbstractController.php [hunk 2]
```diff
@@ -15,6 +16,7 @@
 
 abstract class AbstractController implements ViewContextInterface
 {
+    protected static ?string $controllerName = null;
```
Comment (samdark) [inline src/Controller/AbstractController.php:19]: It is controller so it makes sense to rename it to just name.

File: src/Controller/AbstractController.php [hunk 3]
```diff
@@ -74,7 +76,21 @@ private function findLayoutFile(string $file): string
...
+     * @see Inflector::camel2id()
      */
-    abstract protected function name(): string;
+    protected static function getName(): string
+    {
+        if (static::$controllerName !== null) {
+            return static::$controllerName;
+        }
+
+        $name = preg_replace('/(?:.*\\\)([a-z]+)(controller)/iu', '$1', static::class);
```
Comment (samdark) [inline src/Controller/AbstractController.php:91]: Does it take namespace into account? What would be name for app Site Controller?

File: src/Controller/AbstractController.php (hunk 1 above)
Comment (xepozz) [reply src/Controller/AbstractController.php]: Yes sure

File: src/Controller/AbstractController.php (hunk 3 above)
Comment (xepozz) [reply src/Controller/AbstractController.php]: I wrote some words about algroithm [here](https://github.com/yiisoft/app/blob/128ba3fb4fcba2f81050a4aa060b1f6615f626fc/src/Controller/AbstractController.php#L79)

**Comment by xepozz** [comment]
This PR is blocked by https://github.com/yiisoft/strings/pull/29

File: src/Controller/AbstractController.php (hunk 3 above)
Comment (xepozz) [reply src/Controller/AbstractController.php]: https://regex101.com/r/VYp9z0/1

**Comment by samdark** [comment]
👍

**Pull request merged**